pulsar-broker/src/main/java/org/apache/pulsar/broker/service/AbstractDispatcherSingleActiveConsumer.java

```diff
@@ -81,20 +82,33 @@ public AbstractDispatcherSingleActiveConsumer(SubType subscriptionType, int part
 
     protected abstract void cancelPendingRead();
 
-    protected void pickAndScheduleActiveConsumer() {
+    protected void notifyConsumerGroupChanged(Consumer activeConsumer) {
+        if (null != activeConsumer && subscriptionType == SubType.Failover) {
+            consumers.forEach(consumer ->
+                consumer.notifyConsumerGroupChange(activeConsumer.consumerId()));
```

Comment:
The  `consumerId` is a per-connection identifier (unique within a single TCP connection), but different connections from different clients can very well use the same Id. (The id is just a shortcut to refer to a particular topic/subscription without passing the names each time). 

For consumers on a single subscription, there is no single unique identifier. `consumerName` is generally a random generated string, though there is collision detection (since collisions wont affect the correctness) and it can be manually overridden.

I think the only way is to use the Consumer object identify.